ruby/red-arrow/lib/arrow/slicer.rb

```diff
@@ -162,6 +162,10 @@ def select(&block)
       def reject(&block)
         RejectCondition.new(@column, block)
       end
+
+      def match_substring(substring, ignore_case: false)
+        MatchSubstringCondition.new(@column, substring, ignore_case)
```

Comment:
How about making this reusable for other `match_substring` family functions?

```suggestion
        MatchSubstringFamilyCondition.new("match_substring", @column, substring, ignore_case)
```